### PR TITLE
CDAP-19925 Unit Testing for Connections List Page

### PR DESCRIPTION
**Pull request by adityadivami**
# TITLE
Unit Testing for Connections List Page

## Description
Added Unit Tests for Connections List Page

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [x] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)
https://cdap.atlassian.net/browse/CDAP-19925

## Test Plan
Not Applicable

## Screenshots
Not Applicable

### CODE REVIEW

File: app/cdap/components/GridTable/components/mock/mockDataForGrid.ts [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+                scope: 'SYSTEM',
+              },
+            },
+          },
+        ],
+      },
+      insights: {},
+    },
+  ],
+};
```
Comment (njbriggs) [inline app/cdap/components/GridTable/components/mock/mockDataForGrid.ts]: Newline at end of file

Comment (sivaRajanaDivami) [reply app/cdap/components/GridTable/components/mock/mockDataForGrid.ts]: Done.

File: app/cdap/components/ConnectionList/__test__/ConnectionList.test.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * the License.
+ */
+
+import { act, fireEvent, getByTestId, render, screen } from '@testing-library/react';
+import * as apiHelpers from 'components/Connections/Browser/SidePanel/apiHelpers';
+import * as reducer from 'components/Connections/Create/reducer';
+import { createBrowserHistory as createHistory } from 'history';
+import React from 'react';
+import { Route, Router, Switch } from 'react-router';
+import ConnectionList from '..';
```
Comment (seanfeld) [inline app/cdap/components/ConnectionList/__test__/ConnectionList.test.tsx]: please don't import like this from '..'

File: app/cdap/components/ConnectionList/__test__/ConnectionList.test.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      <Router history={history}>
+        <Switch>
+          <Route>
+            <ConnectionList />
+          </Route>
+        </Switch>
+      </Router>
+    );
+
+    // act(() => {
```
Comment (seanfeld) [inline app/cdap/components/ConnectionList/__test__/ConnectionList.test.tsx]: are these comments necessary?

File: app/cdap/components/GridTable/components/GridHeaderCell/__test__/GridHeaderCell.test.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import { render } from '@testing-library/react';
+import React from 'react';
+import GridHeaderCell from '..';
```
Comment (seanfeld) [inline app/cdap/components/GridTable/components/GridHeaderCell/__test__/GridHeaderCell.test.tsx]: from '../index'

File: app/cdap/components/GridTable/types.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ export interface IParams {
 }
 
 export interface IRecords {
-  [key: string]: string;
+  [key: string]: string | unknown;
```
Comment (seanfeld) [inline app/cdap/components/GridTable/types.ts]: is adding this unknown type necessary?

File: app/cdap/components/ConnectionList/__test__/ConnectionList.test.tsx (hunk 2 above)
Comment (adityadivami) [reply app/cdap/components/ConnectionList/__test__/ConnectionList.test.tsx]: done

File: app/cdap/components/ConnectionList/__test__/ConnectionList.test.tsx (hunk 3 above)
Comment (adityadivami) [reply app/cdap/components/ConnectionList/__test__/ConnectionList.test.tsx]: removed the unncessary comments

File: app/cdap/components/GridTable/components/GridHeaderCell/__test__/GridHeaderCell.test.tsx (hunk 4 above)
Comment (adityadivami) [reply app/cdap/components/GridTable/components/GridHeaderCell/__test__/GridHeaderCell.test.tsx]: done

File: app/cdap/components/GridTable/types.ts (hunk 5 above)
Comment (adityadivami) [reply app/cdap/components/GridTable/types.ts]: resolved

**Review by seanfeld — APPROVED** [review]

**Review by mrahanjam — APPROVED** [review]

**Pull request merged**